qiskit_acqua/operator.py

```diff
@@ -574,16 +591,30 @@ def _eval_with_statevector(self, operator_mode, input_circuit, backend, execute_
                     elif pauli[1].v[qubit_idx] == 1 and pauli[1].w[qubit_idx] == 1:
                         circuit.u3(np.pi, np.pi/2, np.pi/2, q[qubit_idx])
                         # circuits[1+idx].y(q[qubit_idx])
-                circuits.append(circuit)
+
+                if len(circuit) == 0:
```

Comment:
A comment about creating lookup table for circuits w/ and w/o gates.
Since you already have a map to record which one is an empty circuit, you can directly store the circuits into the map rather than creating a list for that.

For those with `-1`, just set it to `None`

So you can have something like:
```python
map_of_circuits = {}
map_of_circuits[idx] = None if len(circuit) == 0 else circuit
```

Then, when you calculate the avg number:
```python
circuit = map_of_circuits[idx]
if circuit is None:
    avg += pauli[0]
else:
   avg += pauli[0] * (simulator_initial_state.T.conj().dot(quantum_state_i))
```